flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/SingleInputGateFactory.java

```diff
@@ -124,7 +124,7 @@ public SingleInputGate create(
 			bufferDecompressor = new BufferDecompressor(networkBufferSize, compressionCodec);
 		}
 
-		SingleInputGate inputGate = new SingleInputGate(
```

Comment:
Are the last three hotfix commits even used in the PR? (Introducing interfaces for gates/channels?)

Reply:
I second this question. On it's own, I think they are doing more harm then good (adding extra code that we would have to maintain).

Reply:
Yes, I added and used test implementations for the extracted interfaces.
IMO, extracting interface is a trade-off (and it often pays off). Particularly in this case, with many tests and unnecessary complex (for tests) class:
```
public SingleInputGateImpl(
    String owningTaskName,
    int gateIndex,
    IntermediateDataSetID consumedResultId,
    final ResultPartitionType consumedPartitionType,
    int consumedSubpartitionIndex,
    int numberOfInputChannels,
    PartitionProducerStateProvider partitionProducerStateProvider,
    SupplierWithException<BufferPool, IOException> bufferPoolFactory,
    @Nullable BufferDecompressor bufferDecompressor) {
```

Reply:
Reverted to minimize changes in this PR.